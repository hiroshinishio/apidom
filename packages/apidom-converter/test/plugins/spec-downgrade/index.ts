import dedent from 'dedent';
import { toValue } from '@swagger-api/apidom-core';
import { expect } from 'chai';

import convert from '../../../src';

describe('converter', function () {
  context('plugins', function () {
    context('spec-downgrade', function () {
      specify('should downgrade OpenAPI specificaiton to 3.0.0', async function () {
        const yamlDefinition = dedent`
        openapi: 3.1.0
        `;
        const openApiElement = await convert(yamlDefinition, '3.1.x');

        expect(toValue(openApiElement)).toMatchSnapshot();
      });
    });
  });
});
