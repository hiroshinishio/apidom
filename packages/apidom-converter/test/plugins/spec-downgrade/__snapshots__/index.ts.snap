// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter plugins spec-downgrade should downgrade OpenAPI specificaiton to 3.0.0 1`] = `
Object {
  openapi: 3.0.0,
}
`;
